cmd/queue/main.go

```diff
@@ -243,40 +226,40 @@ func handler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, handler http.H
 }
 
 // Sets up /health and /wait-for-drain endpoints.
-func createAdminHandlers() *http.ServeMux {
+func createAdminHandlers(p *readiness.Probe) *http.ServeMux {
 	mux := http.NewServeMux()
-	// TODO(@joshrider): temporary change while waiting on other PRs to merge (See #4014)
-	mux.HandleFunc(requestQueueHealthPath, healthState.HealthHandler(probeUserContainer, true /*isNotAggressive*/))
+
+	mux.HandleFunc(requestQueueHealthPath, healthState.HealthHandler(p.ProbeContainer, p.IsAggressive()))
 	mux.HandleFunc(queue.RequestQueueDrainPath, healthState.DrainHandler())
 
 	return mux
 }
-
-func probeQueueHealthPath(port int, timeout time.Duration) error {
+func probeQueueHealthPath(port int, timeoutSeconds int) error {
 	url := fmt.Sprintf(healthURLTemplate, port)
-
+	timeoutDuration := readiness.PollTimeout
+	if timeoutSeconds != 0 {
+		timeoutDuration = time.Duration(timeoutSeconds) * time.Second
+	}
 	httpClient := &http.Client{
 		Transport: &http.Transport{
 			// Do not use the cached connection
 			DisableKeepAlives: true,
 		},
-		Timeout: timeout,
+		Timeout: timeoutDuration,
 	}
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutDuration)
+	defer cancel()
+	stopCh := ctx.Done()
 
 	var lastErr error
-
-	// The 25 millisecond retry interval is an unscientific compromise between wanting to get
-	// started as early as possible while still wanting to give the container some breathing
-	// room to get up and running.
-	timeoutErr := wait.PollImmediate(25*time.Millisecond, timeout, func() (bool, error) {
+	timeoutErr := wait.PollImmediateUntil(aggressivePollInterval, func() (bool, error) {
```

Comment:
Though I know Matt suggested it I liked the previous version more, it's shorter :)
🤷‍♀

Reply:
What'd I suggest?